RULES/cc/cc.go

```diff
@@ -204,11 +178,33 @@ func (lib Library) Build(ctx core.Context) {
 
 // CcLibrary for Library returns the library itself, or a toolchain-specific variant
 func (lib Library) CcLibrary(toolchain Toolchain) Library {
-	toolchain = toolchainOrDefault(toolchain)
+	if toolchain == nil {
+		core.Fatal("CcLibrary() called with nil toolchain.")
+	}
+
+	if lib.Out == nil {
```

Comment:
This is a copy-paste of a check from `Library.build()`. In the future if this becomes a bigger issue we can force users to use constructors (`MakeLibrary()`) where we check inputs.